### ISSUE TITLE
fail to send schema request to TiDB

### ISSUE THREAD

**Issue by cosven** [tikv/pd#3340]
### What did you do?

create database with special charactor
```sql
create database `#mysql50#mysqltest-2`
```

### What did you expect to see?
no error log

### What did you see instead?

```
[2020/12/30 10:37:48.636 +00:00] [WARN] [client.go:103] ["SendRequest failed"] [uri=http://127.0.0.1:42989/schema/#mysql50#mysqltest-2] []
[2020/12/30 10:37:48.636 +00:00] [ERROR] [tidb_requests.go:77] ["fail to send schema request to TiDB"] [error="error.tidb.client_request_failed: Request failed with status code 404 from TiDB API: 404 page not found\n"]
```

### What version of PD are you using (`pd-server -V`)?
nightly version

**Comment by CrazyNash** [comment]
hit the same issue from 4.0.8

**Comment by HunDunDM** [comment]
This error occurred in `Key Visualizer` and has been fixed by [tidb-dashboard#840](https://github.com/pingcap/tidb-dashboard/pull/840). PD `mater` / `v5.0.0-ga` / `v4.0.12` will use the repaired tidb-dashboard.
